test/test_basebackup.py

```diff
@@ -625,12 +625,13 @@ def test_handle_site(self, pghoard):
 
         # adjust basebackup interval to be slightly longer than what this
         # basebackup took and make sure it's not retriggered
-        site_config["basebackup_interval_hours"] = (time.monotonic() - backup_start + 1) / 3600
+        backup_duration = max((time.monotonic() - backup_start), 1)
+        site_config["basebackup_interval_hours"] = (backup_duration * 2) / 3600
```

Comment:
I get the sleep below, but I don't get the change here: shouldn't this actually stay at `(backup_duration +1 ) / 3600`, so that is it basically immediately started after the last?

Another thing: if the waiting is a problem, why make the max already here and not only in sleep?

Reply:
Since we launch other backups after the first one, we don't want them being just a bit longer to fail the test, because the test failing is not actually the following one (having the backup start, at line 636) but the one at line 641.
What happened was: 
 - we start a backup, take it's duration + 1 as the interval between backups.
 - set the backup interval, check that a backup is not started immediately
 - wait some time, and start a backup, check that it's started (*)
 - wait for it to finish, immediately call handle site and check we do not start a backup immadiately. ("this must not start a new one" comment in the test).
 Now, if the time elapsed since the backup marked with a (*) takes longer than initial_backup_duration + 1, the new one would be started.
 By padding it to "2 times the initial backup, at least 2 seconds" it leaves us enough buffer to be much less at risk of having the new backup started when we don't expect it to. 
 I hope that's clearer, and if you have ideas how to phrase that concisely in a comment :-)

Reply:
ok.

Would it help to set the config (after we verified that one backup has started) to something big so we do not run into this situation?

Reply:
I tried to stay true to the original intent of the test, but maybe that would help too.

Reply:
To be honest, I still haven'T really understood whats going on here. If changing the config makes this test more deterministic: please go for it. If not, I will take your word for it that this makes this test better. :-)

Reply:
Ok so let's apply this change and reduce the possibility of having failing tests :)